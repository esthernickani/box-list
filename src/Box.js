import React from "react";

const Box = ( { width, height, backgroundColor, handleRemove, id }) => {
    const boxStyle = {
        width: `${width}px`,
        height: `${height}px`,
        backgroundColor: backgroundColor
    }
    console.log(id)
    const handleBoxRemove = () => handleRemove(id)


    return (
        <div className="Box">
            <div className="Individual Box" style={boxStyle}></div>
            <button onClick={handleBoxRemove}>remove</button>
        </div>

    )
}

export default Box;