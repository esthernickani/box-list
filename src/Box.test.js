import React from "react";
import Box from "./Box";
import { render } from "@testing-library/react";
import { v4 as uuid } from 'uuid';

it("renders without crashing", function() {
    const id = uuid()
    const mockRemove = jest.fn()
    render(<Box width={20} height={20} backgroundColor="red" key={id} id={id} handleRemove={mockRemove} />)
})

it("matches snapshot", function() {
    const id = uuid()
    const mockRemove = jest.fn()
    
    const { asFragment } = render(<Box width={20} height={20} backgroundColor="red" key={id} id={id} handleRemove={mockRemove} />)
    expect(asFragment()).toMatchSnapshot()
})