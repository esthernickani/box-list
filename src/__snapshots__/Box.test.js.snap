// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      style="width: 20px; height: 20px; background-color: red;"
    />
    <button>
      remove box
    </button>
  </div>
</DocumentFragment>
`;
